From 05ab60fb80ec9e1eafb6859f050de32269cdddc2 Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Thu, 5 Mar 2015 09:17:41 +0100
Subject: Add (skip) annotation to gck_object_cache_get_attributes

This works around a bug in vala 0.27.1/git master where it would call
this method in the implementation of Pkcs11Certificate::set_property in
seahorse. This causes an infinite loop as
Pkcs11Certificate is an implementation of the GckObjectCacheInterface,
so calling gck_object_cache_get_attributes() for ::set_property which in
turn calls g_object_set(self, "attributes", value, NULL); is going to
loop.

This can be reverted once vala is fixed.

https://bugzilla.gnome.org/show_bug.cgi?id=745629

diff --git a/gck/gck-object-cache.c b/gck/gck-object-cache.c
index 2325bfa..8da15cb 100644
--- a/gck/gck-object-cache.c
+++ b/gck/gck-object-cache.c
@@ -83,7 +83,7 @@ gck_object_cache_default_init (GckObjectCacheIface *iface)
 }
 
 /**
- * gck_object_cache_get_attributes:
+ * gck_object_cache_get_attributes: (skip):
  * @object: an object with an attribute cache
  *
  * Gets the attributes cached on this object.
-- 
cgit v0.10.2

